src/Dapr.Actors/Runtime/ActorStateManager.cs

```diff
@@ -35,17 +36,17 @@ internal ActorStateManager(Actor actor)
             this.defaultTracker = new Dictionary<string, StateMetadata>();
         }
 
-        public async Task AddStateAsync<T>(string stateName, T value, CancellationToken cancellationToken)
+        public async Task AddStateAsync<T>(string stateName, T value, CancellationToken cancellationToken, int? ttlInSeconds)
```

Comment:
Would it make sense to instead add an overload for when the TTL is specified:
`AddStateAsync<T>(string stateName, T value, int ttlInSeconds)` rather than deviate from the .NET guidance that `CancellationToken` be the *last* parameter of a method declaration.  An exception exists to maintain backward compatibility with existing public APIs, but an overload would seem to provide the same effect while maintaining a consistent API.

Also, is it important for TTL to be specified in integer seconds as opposed to, say, a `TimeSpan`, which might be more usable for larger values?  (What do we expect the typical TTL to be?)

Reply:
Right. I was think more of the actual values used for TTLs.  For example, the samples specify a TTL of `360`.  Would that be better expressed as `TimeSpan.FromMinutes(3)` so that users need not do the math (and potentially get it wrong), specially if they have much larger TTLs (e.g. hours/days).

Reply:
Gotcha, yes that makes sense I'll add that in.